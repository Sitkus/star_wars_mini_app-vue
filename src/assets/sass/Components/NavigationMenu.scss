.nav {
    display: flex;
    padding: 0 2rem 2rem;
    margin-bottom: 2rem;
    border-bottom: 1px solid $gray;

    &__link {
        display: flex;
        align-items: center;
        justify-content: space-between;
        padding: 0.2rem 0.5rem;
        font-weight: 400;
        margin-right: 1rem;

        &:hover {
            background-color: $yellow;
        }
    }

    &__link-last {
        margin-right: 0rem;
    }
}

.link-title {
    padding-right: 1rem;
}

.router-link-exact-active {
    background-color: $yellow;
}

@media screen and (min-width: 768px) {
    .nav {
        display: block;
        height: 100%;
        min-width: 15rem;
        border-bottom: none;
        margin-bottom: 0rem;
        padding: 0rem 3rem;

        &__link {
            margin-bottom: 2rem;
            margin-right: 0rem;
        }

        &__link-last {
            margin-bottom: 0rem;
        }
    }
}
