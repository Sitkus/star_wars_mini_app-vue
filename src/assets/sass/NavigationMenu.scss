.nav {
    height: 100%;
    padding: 0 3rem;
    min-width: 15rem;

    &__link {
        display: flex;
        align-items: center;
        justify-content: space-between;
        margin-bottom: 2rem;
        padding: 0.2rem 0.5rem;
        font-weight: 400;

        &:hover {
            background-color: $yellow;
        }
    }

    &__link-last {
        margin-bottom: 0rem;
    }
}

.link-title {
    padding-right: 1rem;
}

.router-link-exact-active {
    background-color: $yellow;
}
