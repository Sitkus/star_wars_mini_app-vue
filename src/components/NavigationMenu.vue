<template>
    <nav class="nav">
        <router-link class="nav__link" to="/">About</router-link>

        <router-link class="nav__link" to="/films">Films</router-link>
    </nav>
</template>

<script>
export default {
    name: 'NavigationMenu'
};
</script>

<style lang="scss" scoped>
@import '@/assets/sass/NavigationMenu.scss';
</style>
