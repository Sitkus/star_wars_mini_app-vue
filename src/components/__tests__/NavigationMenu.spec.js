import { shallowMount } from '@vue/test-utils';
import NavigationMenu from '../NavigationMenu.vue';

let wrapper;

beforeEach(() => {
    wrapper = shallowMount(NavigationMenu, {
        stubs: ['router-link', 'FontAwesomeIcon']
    });
});

describe('NavigationMenu.vue', () => {
    test('render "About" and "Movies" links', () => {
        const links = wrapper.findAll('[data-testid="link"]');

        expect(links.length).toBe(2);
        expect(links.at(0).text()).toBe('About');
        expect(links.at(1).text()).toBe('Movies');
    });

    test('links have proper attributes "to"', () => {
        const links = wrapper.findAll('[data-testid="link"]');

        expect(links.at(0).element.getAttribute('to')).toBe('/');
        expect(links.at(1).element.getAttribute('to')).toBe('/movies');
    });

    test('HTML structure persists', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });
});
