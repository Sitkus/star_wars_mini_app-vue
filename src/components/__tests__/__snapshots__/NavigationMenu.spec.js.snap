// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationMenu.vue HTML structure persists 1`] = `
<nav class="nav">
  <router-link-stub to="/" data-testid="link" class="nav__link"><span class="link-title">
            About
        </span>
    <fontawesomeicon-stub icon="caret-right"></fontawesomeicon-stub>
  </router-link-stub>
  <router-link-stub to="/movies" data-testid="link" class="nav__link nav__link--last"><span class="link-title">
            Movies
        </span>
    <fontawesomeicon-stub icon="caret-right"></fontawesomeicon-stub>
  </router-link-stub>
</nav>
`;
